lib/elixir/lib/access.ex

```diff
@@ -215,7 +219,7 @@ defmodule Access do
 
   See the implementations for `Map.pop/3` or `Keyword.pop/3` for more examples.
   """
-  @callback pop(term :: t, key) :: {value, t}
+  @callback pop(container, key) :: {value, container} when container: t
```

Comment:
Naming the variable `container` and then making it have type `t` is very confusing. We should probably pick a different name. I also don't think the type should be `t` because `nil` is not accepted here.